services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/SigningProperties.java

```diff
@@ -29,11 +29,14 @@ public enum SigningProperties {
 
     ISSUER_DID("issuerDid", "Did of the issuer.", "Provide the DID of the issuer. Needs to match the provided key material.", ProviderConfigProperty.STRING_TYPE, null),
     KEY_ID("keyId", "Id of the signing key.", "The id of the key to be used for signing credentials. The key needs to be provided as a realm key.", ProviderConfigProperty.STRING_TYPE, null),
+    KID_HEADER("kidHeader", "Kid to be set for the JWT.", "The kid to be set in the jwt-header. Depending on the did-schema, the pure key-id might not be enough and can be overwritten here.", ProviderConfigProperty.STRING_TYPE, null),
     PROOF_TYPE("proofType", "Type of the LD-Proof.", "The type of LD-Proofs to be created. Needs to fit the provided signing key.", ProviderConfigProperty.STRING_TYPE, null),
     ALGORITHM_TYPE("algorithmType", "Type of the signing algorithm.", "The type of the algorithm to be used for signing. Needs to fit the provided signing key.", ProviderConfigProperty.STRING_TYPE, Algorithm.RS256),
 
     TOKEN_TYPE("tokenType", "Type of the token.", "The type of the token to be created.  Will be used as `typ` claim in the JWT-Header.", ProviderConfigProperty.STRING_TYPE, "JWT"),
-    DECOYS("decoys", "Number of decoys to be added.", "The number of decoys to be added to the SD-JWT.", ProviderConfigProperty.STRING_TYPE, 0);
+    DECOYS("decoys", "Number of decoys to be added.", "The number of decoys to be added to the SD-JWT.", ProviderConfigProperty.STRING_TYPE, 0),
+    HASH_ALGORITHM("hashAlgorithm", "Hash algorithm for SD-JWTs.", "The hash algorithm to be used for the SD-JWTs.", ProviderConfigProperty.STRING_TYPE, "sha-256"),
```

Comment:
Q: Is it possible to rather default to `sha-512` ? Or is it some specification requirement to default to `sha-256` ?

FYI. Keycloak recently switched it's internal keys to use sha-512 based algorithms for various things (like pbkdf2-sha512 is now used for password hashing instead of pbkdf2-sha256 and also internal tokens are now signed with HS512 instead of HS256).

Reply:
According to SD-JWT specification, the default hash algorithm is sha-256 (https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-07.html#section-5.1.1).

By other [section](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-07.html#name-choice-of-a-hash-algorithm), it might be OK to use sha-512 instead of sha-256.

Reply:
Since we set the _sd_alg claim, it would be ok to use sha-512. However, the default in the spec is 256 and I fear that not all wallets and verifiers do support more than the default(yet). Thus I would prefer to keep 256 as the default here.

Reply:
@wistefan @tnorimat Sounds good, Thanks for the clarifications!